### PR TITLE
Add fuzz targets for lz4-compress crate

### PR DESCRIPTION
**Pull request by killercup**
Created just for fun, after I managed to get cargo fuzz to work on macOS (using a docker-machine VM running a container based on `jimmycuadra/rust` FYI).

Will let this for a few hours, and see if it finds anything :)

(Locally, I also added `-C opt-level=3` to `RUSTFLAGS` for what it's worth)

### CODE REVIEW

File: lz4-compress/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "lz4-compress-targets"
+version = "0.0.0"
+publish = false
+
+[dependencies]
+libfuzzer-sys = { git = "https://github.com/rust-fuzz/libfuzzer-sys.git" }
+lz4-compress = "0.1.0"
```
Comment (frewsxcv) [inline lz4-compress/Cargo.toml]: Would you mind changing this to the git repository?

See https://github.com/rust-fuzz/targets/pull/18

File: Cargo.toml [hunk 2]
```diff
@@ -10,4 +10,5 @@ members = [
     "serde_json",
     "tar",
     "xml-rs",
+    "lz4-compress",
```
Comment (frewsxcv) [inline Cargo.toml]: Nit: Trying to keep this list alphabetized

**Comment by frewsxcv** [comment]
Awesome, thanks!

**Comment by killercup** [comment]
@frewsxcv amended the commit, and also add an `if let Ok(_) …` to get rid of the warning.

(Is there a need to use a black box or something to prevent LLVM from removing the unused computation, btw? Seeing it use 470MB RAM it seems to work as expected, but one can never be too sure.)

(CI failure is because of <https://github.com/rust-lang/cargo/issues/3819>.)

File: lz4-compress/decompress.rs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#![no_main]
+
+#[macro_use] extern crate libfuzzer_sys;
+extern crate lz4_compress;
+
+fuzz_target!(|data: &[u8]| {
+    if let Ok(_) = lz4_compress::decompress(data) {}
```
Comment (frewsxcv) [inline lz4-compress/decompress.rs]: Is there something wrong with doing:

```rust
let _ = lz4_compress::decompress(data);
```

Is there a concern it'd be optimized away?

Comment (killercup) [inline lz4-compress/decompress.rs]: The concern this there (https://github.com/rust-fuzz/targets/pull/40#issuecomment-285976312) but I only did it this way to easily add a `print!(".")` for debugging :)

**Comment by frewsxcv** [comment]
> (Is there a need to use a black box or something to prevent LLVM from removing the unused computation, btw? Seeing it use 470MB RAM it seems to work as expected, but one can never be too sure.)

I haven't ever encountered a case where it optimized away unused computations, but I've only been fuzzing binaries that have been compiled in debug (not release) mode. Though, considering that [the libfuzzer documentation](http://llvm.org/pre-releases/3.7.0/rc2/docs/LibFuzzer.html) encourages fuzzing with different levels of optimization, I'm not too concerned.

**Pull request merged**